sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/CustomizedHeadersTest.java

```diff
@@ -0,0 +1,133 @@
+
+        String customRequestIdHeaderName = "x-ms-client-custom-request-id";
+        String clientProvidedRequestId = UUID.randomUUID().toString();
+
+        Supplier<HttpHeaders> requestIdSupplier = () ->
+            new HttpHeaders().put(customRequestIdHeaderName, clientProvidedRequestId);
+
+        HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(new NoOpHttpClient() {
+                String firstRequestId = null;
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    if (firstRequestId != null) {
+                        String newRequestId = request.getHeaders().getValue(REQUEST_ID_HEADER);
+                        Assertions.assertNotNull(newRequestId);
+                        Assertions.assertNotEquals(newRequestId, firstRequestId);
+                    }
+                    firstRequestId = request.getHeaders().getValue(REQUEST_ID_HEADER);
+                    if (firstRequestId == null) {
+                        Assertions.fail();
+                    }
+                    return Mono.just(mockResponse);
+                }
+            })
+            .policies(new CustomHeadersPolicy(requestIdSupplier))
+            .build();
+        pipeline.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
+        pipeline.send(new HttpRequest(HttpMethod.GET, new URL("http://localhost/"))).block();
+    }
+
+
+    @Test
+    public void clientProvidedRequestIdForRetry() throws Exception {
+
+        String customRequestIdHeaderName = "x-ms-client-custom-request-id";
+        String clientProvidedRequestId = UUID.randomUUID().toString();
+
+        Supplier<HttpHeaders> requestIdSupplier = () ->
+            new HttpHeaders().put(customRequestIdHeaderName, clientProvidedRequestId);
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(new NoOpHttpClient() {
+                String firstRequestId = null;
+
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    if (firstRequestId != null) {
+                        String newRequestId = request.getHeaders().getValue(customRequestIdHeaderName);
+                        Assertions.assertNotNull(newRequestId);
+                        Assertions.assertEquals(newRequestId, firstRequestId);
+                        Assertions.assertEquals(newRequestId, clientProvidedRequestId);
+                    }
+                    firstRequestId = request.getHeaders().getValue(customRequestIdHeaderName);
+                    if (firstRequestId == null) {
+                        Assertions.fail();
+                    }
+                    return Mono.just(mockResponse);
+                }
+            })
+            .policies(new CustomHeadersPolicy(requestIdSupplier),
+                new RetryPolicy(new FixedDelay(1, Duration.of(0, ChronoUnit.SECONDS))))
```

Comment:
Nothing is going to be retried as you haven't setup the client to return an error.